### DISCUSSION TITLE
Is there a plan for supporting native iOS and Android?

### DISCUSSION THREAD

**Discussion by johnnynanjiang** [dfinity/sdk#3607] [Feature Requests]
So that the user could have a native mobile experience.

There is an iOS example at https://github.com/dfinity/examples/blob/master/motoko/ios-notifications/README.md

Somehow it is mainly for receiving push notification, and is WebView instead of native iOS UX.

**Comment by domwoe** [comment]
You can build native apps using one of the following client libraries/agents 

- React native with [agent-js](https://agent-js.icp.xyz/agent/index.html)
- [Flutter](https://github.com/AstroxNetwork/agent_dart)
- Native Android with [ic4j](https://github.com/ic4j/ic4j-agent)
- Native iOS with swift lib [ICPKit](https://github.com/kosta-bity/IcpKit)

For a complete overview of all client libraries checkout https://github.com/dfinity/awesome-internet-computer?tab=readme-ov-file#client-libraries-agents

**Reply by johnnynanjiang** [reply]
Awesome response, appreciate your advice and help @domwoe
